lib/models/item.rb

```diff
@@ -0,0 +1,21 @@
+module Shop
+  class Item
+    attr_reader :id, :product_id
+    attr_accessor :quantity
+
+    def initialize(product_id:, quantity:, id:)
+      @id = id
+      @product_id = product_id
+      @quantity = quantity
+    end
+
+    def fetch_product
+      FetchProduct.new.call(product_id)
+    end
+
+    def total_price
+      product = fetch.product
```

Comment:
not `fetch.product` but `fetch_product` looks like you didn't test it, or even run in terminal?